internal/runbits/requirements/requirements.go

```diff
@@ -78,25 +91,38 @@ func executePackageOperation(prime primeable, packageName, packageVersion string
 			}
 		}()
 	} else {
+		out.Notice(locale.Tl("operating_message", "", pj.NamespaceString(), pj.Dir()))
+	}
+
+	switch params.NsType {
+	case model.NamespacePackage, model.NamespaceBundle:
+		if pj == nil {
+			break
+		}
 		language, err := model.LanguageByCommit(pj.CommitUUID())
 		if err == nil {
 			langName = language.Name
-			ns = model.NewNamespacePkgOrBundle(langName, nsType)
+			ns = model.NewNamespacePkgOrBundle(langName, params.NsType)
+		} else {
+			logging.Debug("Could not get language from project: %v", err)
 		}
-		out.Notice(locale.Tl("operating_message", "", pj.NamespaceString(), pj.Dir()))
+	case model.NamespaceLanguage:
+		ns = model.NewNamespaceLanguage()
+	case model.NamespacePlatform:
+		ns = model.NewNamespacePlatform()
 	}
 
-	var validatePkg = operation == model.OperationAdded
+	var validatePkg = params.Operation == model.OperationAdded && (ns.Type() == model.NamespacePackage || ns.Type() == model.NamespaceBundle)
 	if !ns.IsValid() {
```

Comment:
We should assert the NsType here as well (ie. is package or bundle), cause the logic below is irrelevant for platforms or languages.